src/argilla/client/feedback.py

```diff
@@ -0,0 +1,807 @@
+
+    def __del__(self) -> None:
+        """When the dataset object is deleted, delete all the records as well to avoid memory leaks."""
+        if hasattr(self, "__records") and self.__records is not None:
+            del self.__records
+        if hasattr(self, "__new_records") and self.__new_records is not None:
+            del self.__new_records
+
+    def __enter__(self) -> "FeedbackDataset":
+        """Allows the dataset to be used as a context manager."""
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb) -> None:
+        """When the context manager is exited, delete all the records as well to avoid memory leaks."""
+        self.__del__()
+
+    @property
+    def guidelines(self) -> str:
+        """Returns the guidelines for annotating the dataset."""
+        return self.__guidelines
+
+    @guidelines.setter
+    def guidelines(self, guidelines: str) -> None:
+        """Sets the guidelines for annotating the dataset."""
+        if not isinstance(guidelines, str):
+            raise TypeError(f"Expected `guidelines` to be a string, got {type(guidelines)} instead.")
+        self.__guidelines = guidelines
+
+    @property
+    def fields(self) -> List[FeedbackFieldModel]:
+        """Returns the fields that define the schema of the records in the dataset."""
+        return self.__fields
+
+    @property
+    def questions(self) -> List[FeedbackQuestionModel]:
+        """Returns the questions that will be used to annotate the dataset."""
+        return self.__questions
+
+    @property
+    def records(self) -> List["FeedbackItemModel"]:
+        """Returns the all the records in the dataset."""
+        return self.__records + self.__new_records
+
+    def fetch_records(self) -> None:
+        """Fetches the records from Argilla or HuggingFace and stores them locally.
+
+        If the dataset has not been saved in Argilla or HuggingFace, a warning will be
+        raised and the current records will be returned instead.
+        """
+        if not self.argilla_id:
+            warnings.warn(
+                "No records have been logged into neither Argilla nor HuggingFace, so"
+                " no records will be fetched. The current records will be returned"
+                " instead."
+            )
+            return self.records
+
+        # TODO(alvarobartt): create `ArgillaMixIn` and `HuggingFaceMixIn` classes to inherit their specific methods
+        if self.argilla_id:
+            client = rg.active_client()
+            first_batch = client.get_records(id=self.argilla_id, offset=0, limit=FETCHING_BATCH_SIZE)
+            self.__records = first_batch.items
+            total_batches = first_batch.total // FETCHING_BATCH_SIZE
+            current_batch = 1
+            with tqdm(
+                initial=current_batch,
+                total=total_batches,
+                desc="Fetching records from Argilla",
+            ) as pbar:
+                while current_batch <= total_batches:
```

Comment:
Since the `total` is optional (See the [API ref](https://tqdm.github.io/docs/tqdm/#__init__)), we still use the tqdm progress bar and prepare changes for #2917

```suggestion
            current_batch = 1
            with tqdm(
                initial=current_batch,
                desc="Fetching records from Argilla",
            ) as pbar:
                while current_batch <= total_batches:
```

Reply:
Then I'll need to change the whole loop I guess, since I was also using the computed `total_batches` for updating the `offset` on every request to Argilla API

Reply:
Is there any way to retrieve the total number of records from the metrics?